src/electron/go_vpn_tunnel.ts

```diff
@@ -136,10 +136,11 @@ export class GoVpnTunnel implements VpnTunnel {
   }
```

Comment:
(code above) I don't think most of that logic in the GoVpnTunnel constructor is needed. Tun2Socks will never exit unless we explicitly call stop(). You should be able to get rid of `this.tun2socks.onExit = this.tun2socksExitListener;`

Also, that entire setup should happen in `connect()`, not the constructor.

Reply:
Done. I moved some of the logic to `connect`; however, to remain compatible with the `VpnTunnel` this class still needs to support the `onceDisconnected` promise.